contracts/mining.js

```diff
@@ -489,3 +718,30 @@ actions.handleStakeChange = async (payload) => {
     });
   }
 };
+
+actions.handleNftChange = async (payload) => {
+  const {
+    symbol, nft, add, callingContractInfo,
+  } = payload;
+  if (api.assert(callingContractInfo && callingContractInfo.name === 'nft',
+    'must be called from nft contract')) {
+    await findAndProcessAll('mining', 'nftTokenPools', { symbol }, async (tokenPool) => {
+      const pool = await api.db.findOne('pools', { id: tokenPool.id });
+      if (pool.updating.inProgress
+          && pool.updating.tokenIndex === pool.tokenMiners.length
+          && pool.updating.lastId < nft._id) {
+        return;
+      }
+      const typeProperties = pool.nftTokenMiner.typeMap[nft.properties.type];
```

Comment:
The "type" property of an NFT instance could be changed to anything at any time.  What happens if it's not something in your typeMap when it gets delegated?

Reply:
added type check. should be sufficient here.